### PR TITLE
Add mssql.TIMESTAMP type to revtypes

### PR DESCRIPTION
**Pull request by dhirschfeld**

### CODE REVIEW

File: odo/backends/sql.py [hunk 1]
```diff
@@ -99,7 +100,8 @@
     mssql.DATETIMEOFFSET: string,
     mssql.MONEY: float64,
     mssql.SMALLMONEY: float32,
-    mssql.UNIQUEIDENTIFIER: string
+    mssql.UNIQUEIDENTIFIER: string,
+    mssql.TIMESTAMP: bytes_,
```
Comment (llllllllll) [inline odo/backends/sql.py]: shouldn't this be `datetime`?

**Comment by dhirschfeld** [comment]
The `TIMESTAMP` type in SQL Server has nothing to do with time. I'll refrain from posting my thoughts on that design decision...

https://msdn.microsoft.com/en-us/library/ms182776(v=SQL.90).aspx
> *The Transact-SQL timestamp data type is different from the timestamp data type defined in the SQL-2003 standard. The SQL-2003 timestamp data type is equivalent to the Transact-SQL datetime data type.*

> *A nonnullable timestamp column is semantically equivalent to a binary(8) column. A nullable timestamp column is semantically equivalent to a varbinary(8) column.*

Without this mapping odo tries to parse the `TIMESTAMP` value as a datetime which unsurprisingly blows up.
```python
>>> df = odo(T.head(), pd.DataFrame)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-8-79cdd33bc0c2> in <module>()
----> 1 q = odo(T.head(), pd.DataFrame)

C:\Miniconda3\lib\site-packages\odo\odo.py in odo(source, target, **kwargs)
     89     odo.append.append      - Add things onto existing things
     90     """
---> 91     return into(target, source, **kwargs)

C:\Miniconda3\lib\site-packages\multipledispatch\dispatcher.py in __call__(self, *args, **kwargs)
    162             self._cache[types] = func
    163         try:
--> 164             return func(*args, **kwargs)
    165 
    166         except MDNotImplementedError:

C:\Miniconda3\lib\site-packages\blaze\compute\core.py in into(a, b, **kwargs)
    377     result = compute(b, return_type='native', **kwargs)
    378     kwargs['dshape'] = b.dshape
--> 379     return into(a, result, **kwargs)
    380 
    381 

C:\Miniconda3\lib\site-packages\multipledispatch\dispatcher.py in __call__(self, *args, **kwargs)
    162             self._cache[types] = func
    163         try:
--> 164             return func(*args, **kwargs)
    165 
    166         except MDNotImplementedError:

C:\Miniconda3\lib\site-packages\odo\into.py in wrapped(*args, **kwargs)
     41             raise TypeError('dshape argument is not an instance of DataShape')
     42         kwargs['dshape'] = dshape
---> 43         return f(*args, **kwargs)
     44     return wrapped
     45 

C:\Miniconda3\lib\site-packages\odo\into.py in into_type(a, b, dshape, **kwargs)
     51         if dshape is None:
     52             dshape = discover(b)
---> 53     return convert(a, b, dshape=dshape, **kwargs)
     54 
     55 

C:\Miniconda3\lib\site-packages\odo\core.py in __call__(self, *args, **kwargs)
     81 
     82     def __call__(self, *args, **kwargs):
---> 83         return _transform(self.graph, *args, **kwargs)
     84 
     85 

C:\Miniconda3\lib\site-packages\odo\core.py in _transform(graph, target, source, excluded_edges, ooc_types, **kwargs)
    104     for convert_from, convert_to, f, cost in path_proxy:
    105         try:
--> 106             x = f(x, excluded_edges=excluded_edges, **kwargs)
    107         except NotImplementedError as e:
    108             if kwargs.get('raise_on_errors'):

C:\Miniconda3\lib\site-packages\odo\backends\sql.py in select_or_selectable_to_frame(el, bind, dshape, **kwargs)
    753     data = list(map(tuple, rows))
    754     dtype = [(str(c), dtypes[c]) for c in columns]
--> 755     data = np.array(data, dtype=dtype)
    756     res = pd.DataFrame(data)
    757     return res

ValueError: Error parsing datetime string "" at position 0
```

File: odo/backends/sql.py (hunk 1 above)
Comment (dhirschfeld) [inline odo/backends/sql.py]: You'd think so... but no. (see below)

**Comment by dhirschfeld** [comment]
Travis failure seems unrelated.

**Comment by dhirschfeld** [comment]
With this patch:
```python
>>> df = odo(T.head(), pd.DataFrame)
>>> df['RowVersion'].head()
0    b'\x00\x00\x00\x00\x00\x01\x82S'
1       b'\x00\x00\x00\x00\x00\x019&'
2       b"\x00\x00\x00\x00\x00\x019'"
3    b'\x00\x00\x00\x00\x00\x012\x8c'
4       b'\x00\x00\x00\x00\x00\x019('
Name: RowVersion, dtype: object
```

**Comment by llllllllll** [comment]
weird that it is actually bytes, can you add a comment so I don't forget about that. I restarted the build. If it fails again I will look into it but this lgtm.

**Comment by llllllllll** [comment]
Thanks!

**Pull request merged**